File: testdata/fixtures/bench/kotlin/vulnerable/a03_command_injection.kt
// Source: CWE-78 - OS Command Injection via Runtime.exec in Kotlin
// Expected: BATOU-INJ, TAINT
// OWASP: A03:2021 - Injection (Command Injection)

package com.example.util

fun executeCommand(userInput: String) {
    val cmd = "ping -c 1 $userInput"
    val process = Runtime.getRuntime().exec(cmd)
    val output = process.inputStream.bufferedReader().readText()
    println(output)
}

fun main(args: Array<String>) {
    val host = args.firstOrNull() ?: return
    executeCommand(host)
}

File: testdata/fixtures/bench/perl/vulnerable/a03_command_injection.pl
# Source: CWE-78 - OS Command Injection via system() in Perl
# Expected: BATOU-PL
# OWASP: A03:2021 - Injection (Command Injection)

use strict;
use warnings;

my $input = $ARGV[0];
my $cmd = "ping -c 1 " . $input;
system($cmd);
